### PR TITLE
Add texture event

### PR DESCRIPTION
**Pull request by josephtarbit**
Allow modders to change the texture of entities before they're rendered. This can be useful for modifying player skins in game allowing for custom clothing etc.

### CODE REVIEW

File: src/main/java/net/minecraftforge/client/event/RenderLivingEvent.java [hunk 1]
```diff
@@ -69,4 +70,22 @@ public Post(LivingEntity entity, LivingRenderer<T, M> renderer, float partialRen
...
+			this.textureLocation = textureLocation;
+		}
+		
+		public ResourceLocation getTextureLocation() { return textureLocation; }
+		
+		public void setTextureLocation(ResourceLocation textureLocation) 
+		{ 
+			this.textureLocation = textureLocation; 
+		}
+    }
```
Comment (diesieben07) [inline src/main/java/net/minecraftforge/client/event/RenderLivingEvent.java:92]: This indentation is all over the place. 4 spaces should be used.

File: patches/minecraft/net/minecraft/client/renderer/entity/LivingRenderer.java.patch [hunk 2]
```diff
@@ -29,6 +29,17 @@
           f8 = MathHelper.func_219799_g(p_225623_3_, p_225623_1_.field_184618_aE, p_225623_1_.field_70721_aZ);
           f5 = p_225623_1_.field_184619_aG - p_225623_1_.field_70721_aZ * (1.0F - p_225623_3_);
           if (p_225623_1_.func_70631_g_()) {
+@@ -109,6 +112,9 @@
+       boolean flag2 = !flag1 && !p_225623_1_.func_98034_c(Minecraft.func_71410_x().field_71439_g);
+       this.field_77045_g.func_225597_a_(p_225623_1_, f5, f8, f7, f2, f6);
+       ResourceLocation resourcelocation = this.func_110775_a(p_225623_1_);
++      net.minecraftforge.client.event.RenderLivingEvent.Texture<T, M> event = new net.minecraftforge.client.event.RenderLivingEvent.Texture<>(p_225623_1_, this, p_225623_3_, p_225623_4_, resourcelocation);
++      net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(event);
++      resourcelocation = event.getTextureLocation();
```
Comment (diesieben07) [inline patches/minecraft/net/minecraft/client/renderer/entity/LivingRenderer.java.patch]: I don't know what the base for this patch file was, but this creates broken code. There is no `resourceLocation` variable in this method.

**Review by diesieben07 — CHANGES_REQUESTED** [review]

File: patches/minecraft/net/minecraft/client/renderer/entity/LivingRenderer.java.patch (hunk 2 above)
Comment (bl4ckscor3) [reply patches/minecraft/net/minecraft/client/renderer/entity/LivingRenderer.java.patch]: There is one in line 35, though?

Comment (diesieben07) [reply patches/minecraft/net/minecraft/client/renderer/entity/LivingRenderer.java.patch]: Which doesn't exist in the actual code.
This patch changes `func_225623_a_`, which does not use `ResourceLocation` at all. This is what the file looks like after this patch is applied (achieved by checking out the PR "from" branch and running setup):
https://i.imgur.com/ZIQQArI.png

Comment (josephtarbit) [reply patches/minecraft/net/minecraft/client/renderer/entity/LivingRenderer.java.patch]: That's very strange, I'll try cleaning my workspace

**Pull request closed without merging**